testnet-contracts/contracts/BridgeBank/BridgeBank.sol

```diff
@@ -111,6 +111,49 @@ contract BridgeBank is CosmosBank, EthereumBank {
             );
     }
 
+    /*
+     * @dev: Burns received Ethereum funds.
+     *
+     * @param _recipient: bytes representation of destination address.
+     * @param _token: token address in origin chain (0x0 if ethereum)
+     * @param _amount: value of deposit
+     */
+    function burn(bytes memory _recipient, address _token, uint256 _amount)
+        public
+        payable
+        availableNonce()
+    {
+        string memory symbol;
+
+        if (msg.value > 0) {
+            require(
+                _token == address(0),
+                "Ethereum burns require the 'token' address to be the null address"
+            );
+            require(
+                msg.value == _amount,
+                "The transactions value must be equal the specified amount (in wei)"
+            );
+            // Burn Ethereum and set symbol
+            address(this).transfer(_amount);
```

Comment:
question - do you need to transfer the amount to this address explicitly? doesn't the fact that it's payable make that happen automatically?

Reply:
Good catch, removed in https://github.com/cosmos/peggy/pull/190/commits/a1f617d32b05edf2d1e0db9130c6ee09e7ae14e9. We actually shouldn't support burns of Ethereum at all, only BridgeTokens. Removed in https://github.com/cosmos/peggy/pull/190/commits/e44976d2e23808a392fe4ef6feb58dbfdfce049b to prevent users from accidentally burning Ethereum as it should be transferred to Cosmos via lock.